### PR TITLE
Ignore Security Warnings for math/rand

### PR DESCRIPTION
**Pull request by saurami**
Updated test files so that `gosec` ignores `math/rand`.  Files updated based on [this CI job](https://github.com/exercism/go/runs/7734762492?check_suite_focus=true) 

For these exercises `math/rand` is better a better option than `crypto/rand` based on suggestion in PR #2416

### CODE REVIEW

**Review by saurami — COMMENTED** [review]
added explanation for ignoring gosec on math/rand

**Comment by saurami** [comment]
Hi @junedev please review this PR

**Comment by saurami** [comment]
Fixes for PR #2416

**Comment by saurami** [comment]
Add change from PR #2416

**Comment by saurami** [comment]
added Diffie Hellman practice exercise

**Comment by saurami** [comment]
Hi @junedev can you have a look at this ?

I've added a no lint directive to appropriate files, but still getting an error.

I'm unable to clear GitHub runner cache.

Please suggest.

File: exercises/concept/animal-magic/.meta/exemplar.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//nolint:gosec // Randomness is part of an exercise where math.rand is more appropriate than crypto.rand
```
Comment (junedev) [inline exercises/concept/animal-magic/.meta/exemplar.go:1]: I found the phrasing a bit hard to follow and the two comment expressions in one line seemed strange. What do you think about the following instead? (If you like it, please apply it to all the places.)

```suggestion
// In the context of the exercise, it is fine to use math.Rand instead of crypto.Rand.
// The following directive will deactivate the respective linter check.
//nolint:gosec
```

File: exercises/practice/diffie-hellman/diffie_hellman_test.go [hunk 2]
```diff
@@ -1,3 +1,4 @@
+ //nolint:gosec // Randomness is part of an exercise where math.rand is more appropriate than crypto.rand
```
Comment (junedev) [inline exercises/practice/diffie-hellman/diffie_hellman_test.go:1]: Why does the test file need this? I don't see an import for math.rand.

File: exercises/practice/binary-search/binary_search_test.go [hunk 3]
```diff
@@ -1,3 +1,4 @@
+//nolint:gosec // Randomness is part of an exercise where math.rand is more appropriate than crypto.rand
```
Comment (junedev) [inline exercises/practice/binary-search/binary_search_test.go:1]: What do you think about this phrasing instead?

```suggestion
//nolint:gosec // In the context of this exercise, it is fine to use math.Rand instead of crypto.Rand.
```

**Comment by junedev** [comment]
@saurami The bug with CI is really strange. I closed and re-opened the PR to start another run. Sometimes that helps.

**Comment by junedev** [comment]
@saurami I found the issue. There is one space character in front of "//nolint" in the file which CI complains about. If you remove the space, everything should be fine. (I found out because the CI report also contained the message that the code was not formatted correctly.)

File: exercises/practice/diffie-hellman/diffie_hellman_test.go (hunk 2 above)
Comment (saurami) [reply exercises/practice/diffie-hellman/diffie_hellman_test.go]: this was a mistake, I fixed it

File: exercises/practice/binary-search/binary_search_test.go (hunk 3 above)
Comment (saurami) [reply exercises/practice/binary-search/binary_search_test.go]: Thank you! All files have been updated.

**Comment by saurami** [comment]
All changes made based on suggestions

**Review by junedev — APPROVED** [review]
Looks all good now.
